pkg/cluster/nodes/nodes.go

```diff
@@ -99,3 +100,35 @@ func list(visit func(string, *Node), filters ...string) error {
 	}
 	return nil
 }
+
+// WaitForReady uses kubectl inside the "node" container to check if the
+// master nodes are "Ready".
+func WaitForReady(node *Node, until time.Time) bool {
+	return tryUntil(until, func() bool {
+		cmd := node.Command(
+			"kubectl",
+			"--kubeconfig=/etc/kubernetes/admin.conf",
+			"get",
+			"nodes",
+			"--selector=node-role.kubernetes.io/master",
+			// When the node reaches status ready, the status field will be set
+			// to true.
+			"-o=jsonpath='{.items..status.conditions[-1:].status}'",
```

Comment:
like i mentioned in one of the comments the better approach is to watch an expected list of pods, but this is also OK in most cases.

Reply:
the e2e tests already do a variant of this, and it doesn't work properly. the pods exist before CNI is ready.

Reply:
given we are hardcoding weave, we can potentially anticipate it (somehow).